src/engine/ControlHelpers.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Godot;
+
+/// <summary>
+///   Common helper operations for Controls
+/// </summary>
+public static class ControlHelpers
+{
+    /// <summary>
+    ///   Shows the popup in the center of the screen and shrinks it to the minimum size,
+    ///   alternative to PopupCentered.
+    /// </summary>
+    public static void PopupCenteredShrink(this Popup popup)
+    {
+        popup.PopupCentered(popup.GetMinimumSize());
+
+        // In case the popup sizing stuck (this happens sometimes)
+        Invoke.Instance.Queue(() =>
```

Comment:
I'd prefer to add a boolean parameter to this method that defaults to true, called something like runSizeUnstuck. So if something ever needs to skip this code, they can.